src/resources/views/crud/inc/form_save_buttons.blade.php

```diff
@@ -26,7 +26,7 @@
 </div>
 @endif
 @if(!$crud->hasOperationSetting('showCancelButton') || $crud->getOperationSetting('showCancelButton') == true)
-    <a href="{{ $crud->hasAccess('list') ? url($crud->route) : url()->previous() }}" class="btn btn-default"><span class="la la-ban"></span> &nbsp;{{ trans('backpack::crud.cancel') }}</a>
+    <a href="{{ is_callable($crud->getOperationSetting('cancelButtonUrl')) ? $crud->getOperationSetting('cancelButtonUrl')($crud) : is_null($crud->getOperationSetting('cancelButtonUrl')) ? ($crud->hasAccess('list') ? url($crud->route) : url()->previous()) : $crud->getOperationSetting('cancelButtonUrl') }}" class="btn btn-default"><span class="la la-ban"></span> &nbsp;{{ trans('backpack::crud.cancel') }}</a>
```

Comment:
At this point I think it will be better if we moved this to a new blade file - `src/resources/views/crud/buttons/cancel.blade.php` where we can make that logic clearer with a `@php @endphp` block at the beginning of the file.

It has quite a bit of logic, and if anybody needs to customize it, they would just overwrite the `cancel.blade.php` file - easier that way.

Reply:
Other than that, I only foresee one small problem with this - that you can't overwrite this setting from the `config/backpack/crud.php` file with a closure, only from the CrudController. At least that's what _I think_, because `$crud` is a parameter...

Would be interesting to know.